hoomd/AlchemyData.h

```diff
@@ -0,0 +1,160 @@
+
+    void inline zeroForces()
+        {
+        ArrayHandle<Scalar> h_forces(m_alchemical_derivatives,
+                                     access_location::host,
+                                     access_mode::overwrite);
+        memset((void*)h_forces.data, 0, sizeof(Scalar) * m_alchemical_derivatives.getNumElements());
+        }
+
+    void resizeForces(unsigned int N)
+        {
+        GlobalArray<Scalar> new_forces(N, m_exec_conf);
+        m_alchemical_derivatives.swap(new_forces);
+        }
+
+    void setNetForce(uint64_t timestep)
+        {
+        // TODO: remove this sanity check after we're done making sure timing works
+        zeroForces();
+        m_timestepNetForce.first = timestep;
+        }
+
+    void setNetForce()
+        {
+        Scalar netForce(0.0);
+        ArrayHandle<Scalar> h_forces(m_alchemical_derivatives,
+                                     access_location::host,
+                                     access_mode::read);
+        for (unsigned int i = 0; i < m_alchemical_derivatives.getNumElements(); i++)
+            netForce += h_forces.data[i];
+        // TODO: make clear the implementation choices being averaged quantities
+        netForce /= Scalar(m_alchemical_derivatives.getNumElements());
+        m_timestepNetForce.second = netForce;
+        }
+
+    void setNetForce(Scalar norm_value)
+        {
+        setNetForce();
+        m_timestepNetForce.second *= norm_value;
+        }
+
+    Scalar getNetForce(uint64_t timestep)
+        {
+        // TODO: remove this sanity check after we're done making sure timing works
+        assert(m_timestepNetForce.first == timestep);
+        return m_timestepNetForce.second;
+        }
+
+    Scalar getNetForce()
+        {
+        return m_timestepNetForce.second;
+        }
+
+    void setMass(Scalar new_mass)
+        {
+        mass.x = new_mass;
+        mass.y = Scalar(1.) / new_mass;
+        }
+
+    Scalar getMass()
+        {
+        return mass.x;
+        }
+
+    Scalar getValue()
+        {
+        return value;
+        }
+
+    pybind11::array_t<Scalar> getDAlphas()
+        {
+        ArrayHandle<Scalar> h_forces(m_alchemical_derivatives,
+                                     access_location::host,
+                                     access_mode::read);
+        return pybind11::array(m_alchemical_derivatives.getNumElements(), h_forces.data);
+        }
+
+    Scalar momentum = 0.; // the momentum of the particle
+    Scalar2 mass
+        = make_scalar2(1.0, 1.0); // mass (x) and it's inverse (y) (don't have to recompute constantly)
+    Scalar mu = 0.;          //!< the alchemical potential of the particle
+    GlobalArray<Scalar> m_alchemical_derivatives; //!< Per particle alchemical forces
+    protected:
+    // the timestep the net force was computed and the netforce
+    std::pair<uint64_t, Scalar> m_timestepNetForce;
+    };
+
+class AlchemicalPairParticle : public AlchemicalMDParticle
```

Comment:
This is the only particle I'm actually using, however I'm trying to lay down some basic information that would be required for any alchemical method to potentially be used elsewhere when needed